src/monitoring_service/service.py

```diff
@@ -78,7 +78,8 @@ def handle_event(event: Event, context: Context) -> None:
                 sentry_sdk.capture_exception(ex)
 
 
-class MonitoringService:  # pylint: disable=too-few-public-methods,too-many-instance-attributes
+class MonitoringService(gevent.Greenlet):
```

Comment:
Please don't inherit from `Greenlet`! It ruins mypy checks and couples the implementation to gevent.

We didn't settle on gevent guidelines yet, but I suggest something like
```python

ms_greenlet = spawn_named("MonitoringService", ms)
...
ms_greenlet.kill()
```